src/lib/setup.js

```diff
@@ -1,102 +1,40 @@
-import path from 'path';
-import which from 'which';
-import fs from 'fs.promised';
 import spawn from 'cross-spawn-promise';
-import { commandExists } from './shell';
+import { hasCommand, warn } from '../util';
 
-const initialize = async (yarn, cwd) => {
-	let isYarnAvailable = await commandExists('yarn');
+export function install(cwd, isYarn) {
+	let cmd = isYarn ? 'yarn' : 'npm';
+	return spawn(cmd, ['install'], { cwd, stdio:'ignore' });
+}
 
-	if (isYarnAvailable && yarn) {
-		return await spawn('yarn', ['init', '-y'], { cwd, stdio: 'ignore' });
-	}
-
-	await spawn('npm', ['init', '-y'], { cwd, stdio: 'ignore' });
-};
-
-const install = async (yarn, cwd, packages, env) => {
-	let isDev = env === 'dev' ? true : false;
-	let isYarnAvailable = await commandExists('yarn');
-	let toInstall = packages.filter(Boolean);
-
-	// pass null to use yarn only if yarn.lock is present
-	if (!yarn) {
-		try {
-			let stat = await fs.stat(path.resolve(cwd, 'yarn.lock'));
-			yarn = stat.isFile();
-		}
-		catch (e) { yarn = false; }
-	}
-
-	if (isYarnAvailable && yarn) {
-		let args = ['add'];
-		if (isDev) {
-			args.push('-D');
-		}
-
-		return await spawn('yarn', [...args, ...toInstall], { cwd, stdio: 'ignore' });
-	}
+export async function addScripts(obj, cwd, isYarn) {
+	let cmd = isYarn ? 'yarn' : 'npm';
+	let args = isYarn ? ['add', '--dev'] : ['install', '--save-dev'];
 
-	await spawn('npm', ['install', isDev ? '--save-dev' : '--save', ...toInstall], { cwd, stdio: 'ignore' });
-};
-
-const pkgScripts = async (yarn, pkg) => {
-	let isYarnAvailable = await commandExists('yarn');
-
-	if (isYarnAvailable && yarn) {
-		return {
-			...(pkg.scripts || {}),
-			start: 'if-env NODE_ENV=production && yarn run -s serve || yarn run -s dev',
-			build: 'preact build',
-			serve: 'preact build && preact serve',
-			dev: 'preact watch',
-			test: 'eslint src && preact test'
-		};
-	}
+	// Install `if-env`
+	await spawn(cmd, [...args, 'if-env'], { cwd, stdio:'ignore '});
 
 	return {
-		...(pkg.scripts || {}),
-		start: 'if-env NODE_ENV=production && npm run -s serve || npm run -s dev',
 		build: 'preact build',
 		serve: 'preact build && preact serve',
-		dev: 'preact watch',
-		test: 'eslint src && preact test'
+		start: `if-env NODE_ENV=production && ${cmd} run -s serve || ${cmd} run -s watch`,
+		watch: 'preact watch'
```

Comment:
Lol okay. I thought this would make more sense since the command name is `watch`.

Reply:
I waffled on this incessantly when doing the original scripts.  FWIW CRA does neither, they actually use `npm start` for the dev server (that's why I added the env bit there).

I don't have a really compelling reason to go one way or the other here.  It might be annoying for people to upgrade and find the command they know has a new name, but then it's only in the templated projects - anything created with 1.x will work just fine.

Reply:
Yeah, however, it _is_ a major release, so this would be the time to change it, if any.